core/schema/modulesource.go

```diff
@@ -1392,323 +1424,221 @@ func (s *moduleSourceSchema) generatedCodeWithVCSIgnoredPaths(ctx context.Contex
 	return code.WithVCSIgnoredPaths(args.Paths), nil
 }
 
-func (s *moduleSourceSchema) moduleSourceWithDependencies(
-	ctx context.Context,
-	parentSrc *core.ModuleSource,
-	args struct {
-		Dependencies []core.ModuleSourceID
-	},
-) (*core.ModuleSource, error) {
-	parentSrc = parentSrc.Clone()
+// itemType distinguishes between dependencies and toolchains in error messages and field access
+type itemType int
 
-	dag, err := core.CurrentDagqlServer(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get dag server: %w", err)
+const (
+	itemTypeDependency itemType = iota
+	itemTypeToolchain
+)
+
+func (t itemType) String() string {
+	switch t {
+	case itemTypeDependency:
+		return "dependency"
+	case itemTypeToolchain:
+		return "toolchain"
+	default:
+		return "unknown"
 	}
+}
 
-	newDeps, err := collectIDObjectResults(ctx, dag, args.Dependencies)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load module source dependencies from ids: %w", err)
+func (t itemType) Plural() string {
+	switch t {
+	case itemTypeDependency:
+		return "dependencies"
+	case itemTypeToolchain:
+		return "toolchains"
+	default:
+		return "unknowns"
+	}
+}
+
+// itemAccessor provides unified access to dependencies or toolchains in a ModuleSource
+type itemAccessor struct {
+	typ itemType
+}
+
+func (a itemAccessor) getItems(src *core.ModuleSource) []dagql.ObjectResult[*core.ModuleSource] {
```

Comment:
nit: feel like it would be much more straightforward to just have `getItems` and `setItems` be methods on `ModuleSource` like `func (src *ModuleSource) getItems(typ itemType)`